vector/src/main/java/im/vector/app/features/roomdirectory/PublicRoomsController.kt

```diff
@@ -109,7 +115,29 @@ class PublicRoomsController @Inject constructor(private val stringProvider: Stri
         }
     }
 
+    private fun PublicRoomsViewState.buildUnknownRoomIfNeeded(): UnknownRoomItem? {
+        val roomIdOrAlias = currentFilter.trim()
+        val isAlias = MatrixPatterns.isRoomAlias(roomIdOrAlias) && !publicRooms.any { it.canonicalAlias == roomIdOrAlias }
+        val isRoomId = !isAlias && MatrixPatterns.isRoomId(roomIdOrAlias) && !publicRooms.any { it.roomId == roomIdOrAlias }
+        val roomItem = when {
+            isAlias  -> MatrixItem.RoomAliasItem(roomIdOrAlias, roomIdOrAlias)
+            isRoomId -> MatrixItem.RoomItem(roomIdOrAlias)
+            else     -> null
+        }
+        return roomItem?.let {
+            UnknownRoomItem_().apply {
+                id(roomIdOrAlias)
+                matrixItem(roomItem)
```

Comment:
I prefer to use `it` here, instead of `roomItem`

Reply:
good catch